overlord/snapstate/check_snap_test.go

```diff
@@ -1030,6 +1031,18 @@ var systemUsernamesTests = []struct {
 }, {
 	sysIDs: "snap_daemon:\n    scope: shared",
 	scVer:  "dead 2.4.1 deadbeef bpf-actlog",
+}, {
+	sysIDs: "snap_microk8s: shared",
+	scVer:  "dead 2.4.1 deadbeef bpf-actlog",
+	snapID: "some-uninteresting-snap-id",
+	error:  `snap "foo" is not allowed to use the system user "snap_microk8s"`,
+}, {
+	snapID: "EaXqgt1lyCaxKaQCU349mlodBkDCXRcg", // microk8s
+	sysIDs: "snap_microk8s: shared",
+	scVer:  "dead 2.4.1 deadbeef bpf-actlog",
+}, {
+	sysIDs: "snap_microk8s:\n    scope: shared",
```

Comment:
maybe a comment here that the snap id is unset so things are allowed

Reply:
Done